lib/VhostContainer.php

```diff
@@ -150,37 +164,42 @@ public function getDefaultHost(): Vhost {
 
     private function selectHostByAuthority(InternalRequest $ireq) {
         $explicitHostId = "{$ireq->uriHost}:{$ireq->uriPort}";
+
+        if (isset($this->vhosts[$explicitHostId])) {
+            return $this->vhosts[$explicitHostId];
+        }
+
+        if ($this->cachedVhostCount === 1 && isset($this->options) && !$this->options->strictHostMatch) {
```

Comment:
I'm not so sure whether we want to just unconditionally allow it on any single vhost. As far as I see, it actually only makes sense on actual wildcard hosts... (If your host has a name, just remove the name then.)
It looks to me like unconditionally allowing it on non-wildcard hostnames is a bad idea - either you want a name _or_ you want it unconditionally.

Reply:
Matching unconditionally requires having only a single host as well as setting an option, something seemingly difficult to do unintentionally. However I agree – this should really only work for wildcard hosts.

Is there an easy way to check if the single host is a wildcard host or will I have to set up another property or two in the container to keep track of wildcard hosts?

Reply:
There are only two wildcard host variants as you see a few lines below, they start with either `0.0.0.0:` or `[::]:`.

Reply:
@bwoebi What's with wildcard hosts listening on `127.0.0.1` then?